### PR TITLE
WIP: Moved grad_norm tracking code to __run_tng_batch 

### LINKED ISSUE #272: Trainer track_grad_norm always results in 0
**Describe the bug**
The Trainer has a flag `track_grad_norm` which allows us to log the gradient norms to Tensorboard. This flag is checked in the `run_tng_epoch` function after the training step and validation step. However, the training step (`__run_tng_batch`) calls `model.optimizer_step()`, which, in the default implementation, calls `optimizer.zero_grad()`. This results in the tracked gradient norms to be always zero. 

Moreover, there is an optional `__run_evaluation` call in the validatin_step. This results in a call to `model.zero_grad()`, which I assume will also result in zero gradient norms. 

**To Reproduce**
1. Train a model with the `track_grad_norm` flag set to True and Tensorboard logging enabled
2. Go to Tensorboard
3. Check the gradient_norms

**Expected behavior**
The gradient norms should not always be zero.

**Desktop (please complete the following information):**
 - OS: Windows
 - Version 0.4.9

### PR DESCRIPTION
**Pull request by kvhooreb**
## What does this PR do?
Fixes #272 

I had to change `metrics.update(grad_norm_dic)` to `self.__add_tqdm_metrics(grad_norm_dic)` to be able to save the metrics. However, with the current implementation all of the `grad_norm` metrics end up in the tqdm progress bar, which, I assume,  is not what we want. Since I'm new to the code, the specifics of the `tqdm_metrics` attribute are not completely clear to me yet. So, I'd like to hear your thought about how to tackle this. Some ideas:
- We could spit up metrics to be logged to disk and metrics to be shown in tqdm
- Maybe we could move all metrics code (https://github.com/williamFalcon/pytorch-lightning/blob/master/pytorch_lightning/trainer/trainer.py#L1011-L1038) into the new location?
- We could create a new parameter to the function and write metrics to that.
- Others? I may miss something obvious of course.

### CODE REVIEW

**Comment by williamFalcon** [comment]
yeah not ideal to add to prog bar. maybe add directly to self.logger?  it assumes you have a logger in there as well, so maybe check for that and add an error otherwise?
